### PR TITLE
#162688568 Fix redirection to and from the error page

### PR DESCRIPTION
**Pull request by dmaish**
#### What does this PR do?
- This PR fixes the popping of the error page whenever a user submits wrong data when editing a particulare guesthouse's details.

#### Description of Task to be completed?
- Only dispatch the error page when the user tries to access an unexisting URL.also, only notify the user through a toast if they have submitted wrong data during editing of an existing guesthouse(submitting rooms with the same room names in the same guesthouse / editing a guesthouse with a name that already exists).
- Make the editing form resubmitable after the user has received the error toast.So that they can change the data and resubmit again.

#### How should this be manually tested?
- Clone the repo
    `git clone https://github.com/andela/travel_tool_front.git`
 - Check in to the project directory
   `cd travel_tool_front`
 - Checkout the **bg-fix-error-page-bug-#162688568** branch
   `bg-fix-error-page-bug-#162688568`
- login into the app as a super admin
- Navigate to the manage guest houses section and edit a guest house to have an already existing name or two rooms in it that have the same room.
- There should be an error message displayed on a toaster but the user should be able to edit to achieve the necessary information and resubmit the form

#### Any background context you want to provide?
- The error page bug was being caused by the presence of an error page that was being prompted by the existence of an error that came from the reducers i.e if a user has input the wrong form data.The presence of such an error page wasn't justified in the first place as such pages are should be rendered only if there's a wrong URL error.Furthermore, errors from reducers (in the accomodation page) have already been catered for through displaying them to the user as a toast.

#### What are the relevant pivotal tracker stories?

[#162688568](https://www.pivotaltracker.com/story/show/162688568)

#### Screenshots (if appropriate)
before: 

screen shot 2018-12-20 at 14 51 26

now:

screen shot 2018-12-20 at 14 50 25

### CODE REVIEW

File: src/views/Accommodation/FullGuestHouseDetails/index.jsx [hunk 1]
```diff
@@ -221,50 +222,57 @@ export class GuestHouseDetails extends PureComponent {
       modal, loading, isLoading, addmaintenanceRecord, deleteMaintenanceRecord, updateMaintenanceRecord, maintenanceDetails, accommodation } = this.props;
     const { shouldOpen, modalType } = modal;
     const { period } = this.state;
+
+    console.log('error object+++++++++++++++++', accommodation.error);
```
Comment (ssewilliam) [inline src/views/Accommodation/FullGuestHouseDetails/index.jsx]: @dmaish This line should be removed as its not used

File: src/components/Forms/NewAccommodationForm/NewAccommodation.jsx [hunk 2]
```diff
@@ -203,6 +203,7 @@ class NewAccommodation extends PureComponent {
       };
       if (this.validate() && modalType === 'edit accommodation') {
         editAccommodation(guestHouse.id, guestHouseData);
+        this.setState({ isSubmitting: false });
```
Comment (ssewilliam) [inline src/components/Forms/NewAccommodationForm/NewAccommodation.jsx]: @dmaish  the state we setting here on this line is not tested

File: src/views/Accommodation/FullGuestHouseDetails/index.jsx [hunk 3]
```diff
@@ -29,6 +29,7 @@ import { fetchAvailableRooms } from '../../../redux/actionCreator/availableRooms
 import Preloader from '../../../components/Preloader/Preloader';
 import ButtonLoadingIcon from '../../../components/Forms/ButtonLoadingIcon';
 import NotFound from '../../ErrorPages';
+import { Accommodation } from '..';
```
Comment (ssewilliam) [inline src/views/Accommodation/FullGuestHouseDetails/index.jsx]: @dmaish This Accommodation we are importing is not in use

File: src/redux/middleware/accommodationSaga.js [hunk 4]
```diff
@@ -100,7 +100,7 @@ export function* editAccommodationAsync(action) {
   catch(error) {
     const errorMessage = apiErrorHandler(error);
     toast.error(errorMessage);
-    yield put(editAccommodationFailure(errorMessage));
+    yield put(editAccommodationFailure(error));
```
Comment (ssewilliam) [inline src/redux/middleware/accommodationSaga.js]: @dmaish I do not believe this change is required

File: src/views/Accommodation/FullGuestHouseDetails/index.jsx (hunk 1 above)
Comment (dmaish) [reply src/views/Accommodation/FullGuestHouseDetails/index.jsx]: this is fixed.

File: src/components/Forms/NewAccommodationForm/NewAccommodation.jsx (hunk 2 above)
Comment (dmaish) [reply src/components/Forms/NewAccommodationForm/NewAccommodation.jsx]: this is fixed.

File: src/views/Accommodation/FullGuestHouseDetails/index.jsx (hunk 3 above)
Comment (dmaish) [reply src/views/Accommodation/FullGuestHouseDetails/index.jsx]: this is fixed.

File: src/redux/middleware/accommodationSaga.js (hunk 4 above)
Comment (dmaish) [reply src/redux/middleware/accommodationSaga.js]: this is fixed.

**Pull request closed without merging**